doc/man3/PKCS12_newpass.pod

```diff
@@ -17,6 +17,16 @@ PKCS12_newpass() changes the password of a PKCS12 structure.
 B<p12> is a pointer to a PKCS12 structure. B<oldpass> is the existing password
 and B<newpass> is the new password.
 
+=head1 NOTES
+
+Both B<oldpass> and B<newpass> are NUL-terminated strings in the UTF-8 encoding.
+If either is not a valid UTF-8 byte sequence, then that one is interpreted
+as ISO8859-1 instead.
+
+In particular, this means that passwords obtained through the default
+UI_OpenSSL() method, which are in the locale character set, should not be
+used without potentially being converted to UTF-8 first.
```

Comment:
That was implicit in the "potentially". And no, not if it's known to be ISO8859-1. That's why we have the hack in `apps/pkcs12.c`, because if it's ISO8859-1 and just *happens* to be a byte sequence that would have been valid UTF-8, it'll get interpreted as UTF-8.